src/hw/hw_mngr.cpp

```diff
@@ -13,10 +13,108 @@
 
 #include <charconv>
 #include <filesystem>
+#include <iostream>
 
 namespace fs = std::filesystem;
 using namespace phosphor::logging;
 
+namespace {
+    constexpr int maxErrorAttempts = 20;
+    constexpr const char* dbusPathFanSensorPwm = "/xyz/openbmc_project/sensors/fan_pwm/";
+    constexpr const char* dbusPathFanSensorRpm = "/xyz/openbmc_project/sensors/fan_tach/";
+    constexpr const char* dbusFanSensorService = "xyz.openbmc_project.FanSensor";
+    constexpr const char* dbusInterfaceSensorValue = "xyz.openbmc_project.Sensor.Value";
+    constexpr const char* dbusPropertyNameValue = "Value";
+    constexpr const char* emptyString = "";
+
+
+    struct {
+        const uint32_t inletRangeMin;
+        const uint32_t inletRangeMax;
+        const uint32_t outletRangeMin;
+        const uint32_t outletRangeMax;
+        const char* partNumber;
+    }
+    detectionFanTable[] = {
+        { 19000, 24000, 16200, 19800, "ASMFAN783104A"  },
+        { 26550, 34000, 22950, 30800, "ASMFAN781104A1" }
+    };
+
+
+    std::string getFanPartNumberByRpm(uint32_t inletRpm, uint32_t outletRpm)
+    {
+        size_t  numItems = sizeof(detectionFanTable) / sizeof(detectionFanTable[0]);
```

Comment:
I can't remember why it's bad practice, but I read about it some time ago.
Likely it was explained as an improved portability.
For the such cases more preferable is the using of one STL containers.
You could use `std::array<...>` or `std::vector<....>` for `detectionFanTable` and then:
```cpp
for (const auto& item : detectionFanTable)
{
    ...
}
```